main.go

```diff
@@ -91,14 +91,13 @@ func main() {
 
 	printVersion()
 
-	namespace, err := getWatchNamespace()
```

Comment:
The only way I can think to support single and cluster deployment modes is to have separate a csv, and config/manager/manager.yaml for each deployment type . As the namespace is set to a blank string in both the csv and manager.yaml to enable cluster mode

The original logic would probably stand up 
```go
	namespace, err := getWatchNamespace()
	if err != nil {
		setupLog.Error(err, "Failed to get watch namespace")
		os.Exit(1)
		//os.Exit(1)
	}

	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
		Namespace:          namespace,
		Scheme:             scheme,
		MetricsBindAddress: metricsAddr,
		Port:               9443,
```

Reply:
I do not think multiple CSV is a feasible option. 

In this [slide](https://docs.google.com/presentation/d/1j1J575SxS8LtL_YvKqrexUhso7j4SgrLfyNrDUroJcc/edit#slide=id.ge0b0895a85_0_185) there is a list of operators supporting multiple install modes. Can you check out how they do? Operator group thing should be involved in this matter, as this component is responsible to fill the annotation `olm.targetNamespaces` used in the CSV to give value to the `WATCH_NAMESPACE` env var.

Reply:
@eguzki it is possible via https://sdk.operatorframework.io/docs/building-operators/golang/operator-scope/#configuring-cluster-scoped-operators-with-multinamespacedcachebuilder to pass in a namespaces comma separated string, will test weather its possible to set one of the namespaces to a empty string

Reply:
I guess the operator  group adds "" (empty) or "ns1" depending on the install mode specified in the operator hub. 

```yaml
---
apiVersion: operators.coreos.com/v1alpha2
kind: OperatorGroup
metadata:
  name: 3scale-operator
  namespace: ns1
spec:
  targetNamespaces:
    - ns1
```

The 3scale operator just needs to read that from the WATCH_NAMESPACE env var and give to the manager via the Options.namespace attribute. The change in this PR is breaking that required behavior

Reply:
We are trying to make the operator cluster scope, not multi namespace scoped. Let's work on the cluster scope and the we can add later the multi namespace scope install mode.

Reply:
Thanks @eguzki had tried updating the operatorGroup in the passed and found it didn't work, I see that the issue is the config/manager/manager.yaml has the WATCH_NAMESPACE defined as metadata.namespace, have removed this env var and it appears to work

Reply:
But we want to support cluster scope and single namespace scope. If you set WATCH_NAMESPACE to `""` it will only be cluster scoped.

Reply:
Thanks @eguzki taught I had revert the WATCH_NAMESPACE env var in the csv to 
```yaml
                - name: WATCH_NAMESPACE
                  valueFrom:
                    fieldRef:
                      fieldPath: metadata.annotations['olm.targetNamespaces']
``` but I hadn't. done now

Reply:
the csv is autogenerated from manager.yaml and other sources.

Please, run `make bundle` to check there csv you committed is correct. I do not think that removing the WATCH_NAMESPACE from the manager.yaml does any good

Reply:
@eguzki correct the WATCH_NAMESPACE env is now removed from the csv , its also reverted the installType, which I believe is expected and should be set after `make bundle` is run.

Reply:
reviewing some cluster scoped operators I have found
- the quay-operator doesn't set WATCH_NAMESPACE env in manager.yaml ns and in the csv. It sets the namespace to an empty string https://github.com/quay/quay-operator/blob/master/main.go#L67
- AMQ online/enmass sets the namespac via NAMESPACE env in the csv https://github.com/EnMasseProject/enmasse/blob/master/olm-manifest/src/main/resources/enmasse.clusterserviceversion.yaml#L539-L542 and namespace is populated in main.go https://github.com/EnMasseProject/enmasse/blob/master/cmd/controller-manager/main.go#L86 so it only changes the installmode to allNamespaces

From testing with 3scale operator once the namespace is populated in the main.go the operator will not act on cr's in other namespaces. So as far as I can tell `AMQ online` is cluster scoped in name only.

Reply:
The quay-operator manages the watched namespace from the command options instead of the env vars. In the CSV there is a `--namespace` with a value taken from the WATCH_NAMESPACE env var, which is in turn read from annotation set by the operator group. https://github.com/quay/quay-operator/blob/caa8303c2a3f1823642cc70d61acd1ed44383441/bundle/downstream/manifests/quay-operator.clusterserviceversion.yaml#L173 It is tricky, but I guess it works. 

Regarding the EnMasse operator, it seems to be namespace scoped, as you say as the manager is always initialized with the current namespace value. 

I think that in the 3scale operator, regarding the code in the `main.go`, we are good to support namespaced and cluster scoped installation modes. At least for initializing the manager. Other issue, to be tested, is the deployment of 3scale. The operator was implemented with namespace scope mindset and maybe some code must be fixed. For instance, the tenant controller does not respect the `v1.SecretReference` namespace value and always uses the Tenant CR namespace value. This is just an example, and does not even affect 3scale deployment, but still a valid example. FYI The tenant controller is part of application capabilities, as set of CRDs and controllers to manage an *existing* 3scale instance. They are not meant to deploy 3scale as the APIManager CRD.